src/eterna/pose2D/Pose2D.ts

```diff
@@ -995,14 +1030,49 @@ export default class Pose2D extends ContainerObject implements Updatable {
                 this.callAddBaseCallback(cmd[0], cmd[1], closestIndex);
             }
 
-            e.stopPropagation();
+            e?.stopPropagation();
         } else if (shiftDown && !this.annotationManager.getAnnotationMode()) {
             this._shiftStart = -1;
             this._shiftEnd = -1;
             this.updateShiftHighlight();
         }
     }
 
+    public onVirtualPoseMouseDown(closestIndex: number): void {
```

Comment:
Why were these "virtual" methods created? This is an awful lot of duplicated code